File: Practice/atcoder/ABC/068/src/c.rb
# そもそもNにいくやつ以外はいらないよねって発想
n, m = gets.chomp.split.map(&:to_i)
a2c, b2c = Array.new(n, false), Array.new(n, false)
m.times do
  a, b = gets.chomp.split.map(&:to_i)
  if b==n
    b2c[a-1] = true
  end
  if a==1
    a2c[b-1] = true
  end
end
n.times do |i|
  if a2c[i] == true && b2c[i] == true
    puts "POSSIBLE"
    exit
  end
end
puts "IMPOSSIBLE"

File: Practice/atcoder/ABC/068/src/c.py
# 1からNへの経由地点Kの気持ちでみると各クエリで判別可能
n, m = map(int, input().split())
arr = [[False, False] for _ in range(n)]
for _ in range(m):
    a, b = map(int, input().split())
    if a == 1:
        arr[b - 1][0] = True
    if b == n:
        arr[a - 1][1] = True

for x in arr:
    if x[0] and x[1]:
        print("POSSIBLE")
        exit()
print("IMPOSSIBLE")
